### DISCUSSION TITLE
Packet Sender for Android

### DISCUSSION THREAD

**Discussion by dannagle** [dannagle/PacketSender#219] [General]
:fireworks: :bell: Ring in the new year!  I'm very excited to announce the from-scratch complete rewrite and new release of Packet Sender for Android:
https://play.google.com/store/apps/details?id=com.packetsender.compose

https://www.amazon.com/dp/B08RXM6KM2/

This new version was written in classic Kotlin and Android Studio following Google's best practices and design principles. I hope you find this app a worthy successor to the old Xamarin-based version.

**Comment by erhome00** [comment]
Hi there,

I'd really love to have a look at the android version of this grate tool, but unfortunately the app store tells me, that my Samsung Galaxy S10 is not supported. Would you tell me if this is a bug with the app store or build configuration, or what devices are supported then?

Thanks in advance!

Best regards,
Erik

**Reply by dannagle** [reply]
Hello. There were reports of the keyboard not appearing for a user with a Samsung S10. Asking the user to switch to GBoard or SwiftKey seemed to fix this, but then later crash reports with no obvious fix appeared. 

I don't have an S10 to test on, so I turned off support for that particular phone model. 

I used an _extremely_ new technique, called Jetpack Compose, to write the app:
https://developer.android.com/jetpack/androidx/releases/compose-ui

Since publishing, there have been a few more releases of Jetpack Compose, so perhaps that bug is fixed now. My current strategy is to set all blocked phones back to being supported once Jetpack Compose reaches full 1.0.0 status. If you are willing, I would be happy to send you the apk built using these latest libraries in exchange that you run it through some basic tests on your phone and then let me know if it is OK to turn Samsung S10 back on early.

**Reply by erhome00** [reply]
Hi again,

thanks for the detailed and fast answer and explanation. I'd be happy to test a new build on my mobile and report if it is working or not. Since you have my email address already, feel free to send me the apk.

Looking forward to test the app on my mobile a bit. I really love the desktop version and had some use cases in the past where it was very helpful.

Thanks again!

Regards,
Erik

**Reply by BahaeddinSagar** [reply]
Hello, 

is there any update on the Android app ? can I be of service in someway ?

Regards

**Reply by dannagle** [reply]
It's still very much broken. Jetpack Compose is now stable, but there are some painful API changes. I just don't have the time.

If you (or anybody) is willing to fix the build (upgrade everything to the latest Jetpack Compose), I will compile and sign a package and make it 100% free (no ads or tracking) on Google Play.

**Reply by BahaeddinSagar** [reply]
well, I am not exactly an expert, but I can attempt on fixing it if you would let me.
